frontend/src/components/nodes/ValidatorRow.tsx

```diff
@@ -44,32 +45,91 @@ const WarningText = styled("td", {
 });
 
 interface Props {
-  node: ValidationNodeInfo;
+  validator: ValidatorFullData;
   index: number;
   totalStake: BN;
   cumulativeStake: BN;
   isNetworkHolder: boolean;
+  seatPrice?: string;
 }
 
 const ZERO = new BN(0);
 const EXTRA_PRECISION_MULTIPLIER = Math.pow(10, 2 + FRACTION_DIGITS);
 
+const getStakingStatus = (
+  validator: ValidatorFullData,
+  seatPrice?: string
+): StakingStatus | null => {
+  if (validator.currentEpoch) {
+    if (validator.nextEpoch) {
+      return "active";
```

Comment:
We have a few of types based on string discrimination, some of them are based on `PascalCase` we get from RPC, the rest is arbitrary.

The only thing I feel uncomfortable is to have a mapping with CAPS keys.

Why did you suggest that? Because of `on-hold` change to `onHold`, I presume?